### PR TITLE
BIP User Activated Soft Fork Split Protection

### PR DESCRIPTION
**Pull request by jameshilliard**
Mailing list discussion: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2017-June/014508.html

### CODE REVIEW

**Comment by kek-coin** [comment]
The SPLITPROTECTION part of the reference implementation does not seem to match the described behaviour; is this still "todo"?

**Comment by jameshilliard** [comment]
@kek-coin What part specifically does not match?

**Comment by kek-coin** [comment]
I'm not seeing the 65% or the 504 block-interval anywhere, and it is not clear to me (from the text or the code) if this is supposed to be a rolling window or a quarter of a difficulty retarget. It seems to refer to a regular BIP9 state machine, and only enforces mandatory signalling during the LOCKED_IN phase; not during the ACTIVE phase.

**Comment by jameshilliard** [comment]
[This](https://github.com/bitcoin/bitcoin/compare/0.14...jameshilliard:splitprotection-v0.14.1#diff-d6954440f0346f8e42dcd669dd4aeafaR158) is the interval and [this](https://github.com/bitcoin/bitcoin/compare/0.14...jameshilliard:splitprotection-v0.14.1#diff-d6954440f0346f8e42dcd669dd4aeafaR167) is the percentage.

File: bip-splitprotection.mediawiki [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+<pre>
+// Check if Segregated Witness is Locked In
+bool IsWitnessLockedIn(const CBlockIndex* pindexPrev, const Consensus::Params& params)
+{
+    LOCK(cs_main);
+    return (VersionBitsState(pindexPrev, params, Consensus::DEPLOYMENT_SEGWIT, versionbitscache) == THRESHOLD_LOCKED_IN);
+}
+
+// SPLITPROTECTION mandatory segwit signalling.
+if ( VersionBitsState(pindex->pprev, chainparams.GetConsensus(), Consensus::DEPLOYMENT_SPLITPROTECTION, versionbitscache) == THRESHOLD_LOCKED_IN &&
```
Comment (Vaesper) [inline bip-splitprotection.mediawiki]: Suggest replacing

`VersionBitsState(pindex->pprev, chainparams.GetConsensus(), Consensus::DEPLOYMENT_SPLITPROTECTION, versionbitscache) == THRESHOLD_LOCKED_IN &&`

with

```
(VersionBitsState(pindex->pprev, chainparams.GetConsensus(), Consensus::DEPLOYMENT_SPLITPROTECTION, versionbitscache) == THRESHOLD_LOCKED_IN ||
VersionBitsState(pindex->pprev, chainparams.GetConsensus(), Consensus::DEPLOYMENT_SPLITPROTECTION, versionbitscache) == THRESHOLD_STARTED ) &&
```

**Comment by luke-jr** [comment]
This is just a revision to BIP 91. Please update the correct file.

**Comment by jameshilliard** [comment]
@luke-jr this is intended to be separate since it uses BIP8 style activation logic instead of BIP9 as well as a different threshold and version bit

**Comment by luke-jr** [comment]
Minor implementation details of a young draft. Just change the existing BIP.

**Comment by jameshilliard** [comment]
@luke-jr I disagree that these are minor details, there are different risks associated with the activation methods.

**Comment by luke-jr** [comment]
@jameshilliard I suggest closing this.

**Pull request closed without merging**